src/main/java/com/sforce/ws/transport/JdkHttpStreamingTransport.java

```diff
@@ -0,0 +1,314 @@
+            HashMap<String, String> httpHeaders) throws IOException {
+        return createConnection(config, url, httpHeaders, true);
+    }
+
+    public static HttpURLConnection createConnection(ConnectorConfig config, URL url,
+            HashMap<String, String> httpHeaders, boolean enableCompression) throws IOException {
+
+        if (config.isTraceMessage()) {
+            config.getTraceStream().println("WSC: Creating a new connection to " + url + " Proxy = " +
+                    config.getProxy() + " username " + config.getProxyUsername());
+        }
+
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection(config.getProxy());
+        if (httpHeaders == null || (httpHeaders.get("User-Agent") == null && httpHeaders.get("user-agent") == null)) {
+            connection.addRequestProperty("User-Agent", VersionInfo.info());
+        }
+
+        /*
+         * Add all the client specific headers here
+         */
+        if (config.getHeaders() != null) {
+            for (Entry<String, String> ent : config.getHeaders().entrySet()) {
+                connection.setRequestProperty(ent.getKey(), ent.getValue());
+            }
+        }
+
+        if (enableCompression && config.isCompression()) {
+            connection.addRequestProperty("Content-Encoding", "gzip");
+            connection.addRequestProperty("Accept-Encoding", "gzip");
+        }
+
+        if (config.getProxyUsername() != null) {
+            String token = config.getProxyUsername() + ":" + config.getProxyPassword();
+            String auth = "Basic " + new String(Base64.encode(token.getBytes()));
+            connection.addRequestProperty("Proxy-Authorization", auth);
+            connection.addRequestProperty("Https-Proxy-Authorization", auth);
+        }
+
+        if (httpHeaders != null) {
+            for (Map.Entry<String, String> entry : httpHeaders.entrySet()) {
+                connection.addRequestProperty(entry.getKey(), entry.getValue());
+            }
+        }
+
+        if (config.getReadTimeout() != 0) {
+            connection.setReadTimeout(config.getReadTimeout());
+        }
+
+        if (config.getConnectionTimeout() != 0) {
+            connection.setConnectTimeout(config.getConnectionTimeout());
+        }
+        
+        if (config.isTraceMessage()) {
+            config.getTraceStream().println("WSC: Connection configured to have request properties " + connection.getRequestProperties());
+        }
+
+        return connection;
+    }
+
+    @Override
+    public InputStream getContent() throws IOException {
+    	if (this.contentInputStream != null) {
+    		// send the request content
+            byte[] buffer = new byte[2048];
+            BufferedOutputStream bos = new BufferedOutputStream(this.connection.getOutputStream());
+	        for(int len; (len = this.contentInputStream.read(buffer)) > 0;) {
+	            bos.write(buffer, 0, len);
+	        }
+	        bos.flush();
+        	bos.close();
+    	}
```

Comment:
What exactly is the "streaming" capability that is added by this class? Since the connect() methods return the outputStream() to the caller, why can't the caller implement the above code for buffering?

Reply:
good question. This change came about due to the implementation HttpClientTransport.java, which uses Apache HttpClient instead of JDK Http utilities. Apache HttpClient does not provide a ready-to-use OutputStream to write request to. Instead, it accepts an InputStream to read content from. However, I will cancel this PR and think about a way to support large input on the client side that can be handled by HttpClientTransport with the existing method signatures of Transport.java.